src/components/ChattingScreen.js

```diff
@@ -18,14 +21,96 @@ export default function ChattingScreen() {
 		{ user: false, content: '나도 아니까  ' },
 		{ user: false, content: '그만 말해줘도 돼' },
 		{ user: true, content: 'ㅠㅠ' },
-	];
+	]);
+	function changeSender(e){
+		e.preventDefault();
+		setCurrentSender(!currentSender);
+	}
+	function changeNewMsg(e){
+		e.preventDefault();
+		setNewMsg({user: currentSender, content: e.target.value});
+	}
+	function updateMsglist(e){
+		e.preventDefault();
+		if(newMsg.content === ''){
+			alert('메시지를 입력해주세요');
+			return;
+		}
+		setCurrentMsglist([...currentMsglist, newMsg]);
+		setNewMsg({user: currentSender, content: ''});
+	}
+	let currentSenderName = currentSender ? USER_T_NAME : USER_F_NAME;
+	let currentSenderIMG = currentSender ? USER_T_IMG : USER_F_IMG;
+
+	useEffect(() => {
+		window.scrollBy(0, window.innerHeight);
+	}, [currentMsglist]);
 
 	return (
 		<Wrapper>
-			<Header></Header>
-			<MessageSender></MessageSender>
+			<Header sender = {currentSenderName} senderImg = {currentSenderIMG} changeSender = {changeSender} ></Header>
+			{currentMsglist.map((message) =>
+				<PrintOneMessage sender = {message.user} message = {message.content}/>
+			)}
+			<MessageSender message = {newMsg.content} changeNewMsg = {changeNewMsg} updateMsglist = {updateMsglist} ></MessageSender>
+			<EmptySpace/>
 		</Wrapper>
 	);
 }
 
-const Wrapper = styled.div``;
+function PrintOneMessage({sender, message}) {
+	if(sender) {
+		return (
+			<LeftMessage>
+				<img src={USER_T_IMG} alt="senderImg"/>
+				<div className="content">{message}</div>
+			</LeftMessage>
+		);
+	}
+	else {
+		return(
+			<RightMessage>
+				<img src={USER_F_IMG} alt="senderImg"/>
+				<div className="content">{message}</div>
+			</RightMessage>
+		);
+	}
+}
+
+const EmptySpace = styled.div`
+	height: 70px;
+`;
+
+const Wrapper = styled.div`
+	background-color: rgb(171,193,209);
+`;
+
+const PrintOneMessageStyle = styled.div`
+	display: flex;
+	flex-direction: row;
+	align-items: center;
+	img {
+		weight: 50px;
+		height: 50px;
+		border-radius: 20px;
+		margin: 10px;
+		box-shadow: 0 0 10px rgba(0,0,0,0.2);
+	}
+	.content{
+		display: flex;
+		align-items: center;
+		padding: 0 10px 0 10px;
+		height: 45px;
+		border-radius: 10px;
+		background-color: white;
+		box-shadow: 0 0 15px rgba(0,0,0,0.1);
+	}
+`;
+
+const LeftMessage = styled(PrintOneMessageStyle)`
+	flex-direction: row;
+`;
+
+const RightMessage = styled(PrintOneMessageStyle)`
+	flex-direction: row-reverse;
+`;
```

Comment:
flexbox 사용을 통해서 css만으로 구분하고, 컴포넌트 상속 기능까지... 최고에요 😭

이름 을 요렇게 지어보면 어떨까요?? Left, Right도 좋은데 조금 더 기능?을 반영했으면 좋겠어요 ㅎㅎ
```suggestion
const YourMessage = styled(PrintOneMessageStyle)` // or OpponentMessage
	flex-direction: row;
`;

const MyMessage = styled(PrintOneMessageStyle)`
	flex-direction: row-reverse;
`;
```

Reply:
우와 너무 좋아요 ! 이거 이름 지을 때 고민을 많이 했는데 left, right 보단 your, my로 짓는게 더 좋은 것 같아요!!

Reply:
와 이 코드 엄청 좋네요..!!
저도 활용하고 싶습니다.